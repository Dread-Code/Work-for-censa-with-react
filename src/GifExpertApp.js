import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";


export const GifExpertApp = ({defaulCategories = []}) => {

    const [categories, setCategories] = useState(defaulCategories);

    return (
      <div>
        <h2> Gif Expert App</h2>
        <AddCategory setCategories = { setCategories }/>
        <hr/>
        <ol>
            {
                categories.map(category =>(
                    <GifGrid 
                    category = {category}
                    key = {category}
                    />
                ))
            }
        </ol>
      </div>
    )
  }