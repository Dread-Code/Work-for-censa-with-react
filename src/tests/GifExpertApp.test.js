import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp'

describe('GifExpert Component', () => {
    
    test('Snapshot test', () => {
        
        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('should ', () => {
        let programs = ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defaulCategories = {programs}/>);

        expect(wrapper).toMatchSnapshot(); // creo el snapshot para la prueba
        expect(wrapper.find('GifGrid').length).toBe( programs.length); // compara si renderizara 
        //lo esperado con los programs que se le pase
        
    });

});