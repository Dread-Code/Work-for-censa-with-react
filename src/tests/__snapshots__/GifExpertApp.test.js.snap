// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpert Component Snapshot test 1`] = `
<div>
  <h2>
     Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`GifExpert Component should  1`] = `
<div>
  <h2>
     Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
